source/features/subducting_plate_models/temperature/mass_conserving.cc

```diff
@@ -0,0 +1,338 @@
+        MassConserving::get_temperature(const Point<3> & /*position_in_cartesian_coordinates*/,
+                                        const double depth,
+                                        const double gravity_norm,
+                                        double temperature_,
+                                        const double /*feature_min_depth*/,
+                                        const double /*feature_max_depth*/,
+                                        const WorldBuilder::Utilities::PointDistanceFromCurvedPlanes &distance_from_planes) const
+        {
+
+          const double distance_from_plane = distance_from_planes.distance_from_plane;
+          const double distance_along_plane = distance_from_planes.distance_along_plane;
+          const double depth_slab_surface = distance_from_planes.depth_reference_surface;
+
+          const double seconds_in_year = 60.0*60.0*24.0*365.25;
+          const double plate_age_sec = age_at_trench*1e6*seconds_in_year;  // my --> seconds
+          const double over_plate_age_sec = overriding_plate_age_above_slab*1e6*seconds_in_year;  // my --> seconds
+
+          if (distance_from_plane <= max_depth && distance_from_plane >= min_depth)
+            {
+
+              // 1. Determine initial heat content of the slab based on age of plate at trench
+              //    This uses the integral of the half-space temperature profile
+              const double initial_heat_content = 2*thermal_conductivity*(surface_temperature - potential_mantle_temperature) *
+                                                  std::sqrt(plate_age_sec/(thermal_diffusivity*const_pi));
+
+              //  2. Get Tmin and distance_offset given distance along slab and depth of point on the slab.
+              //  These equations are empirical based on fitting the temperature profiles from dynamic subduction models.
+              double min_temperature = 0.0;
+              double distance_offset = 0.0;
+
+              const double mantle_coupling_length = mantle_coupling_depth/std::sin(shallow_average_dip*const_pi/180.0); //m
+
+              /* Empirical model parameters */
+              double upper_mantle_depth = 660e3; // m
+              double upper_mantle_length = 1000e3; // m
+              /* Distance offset parameters */
+              double offset_distance_min = 25*1e3; // m
+              double offset_distance_max = 35*1e3; // m
+              double slope_distance_min = 25e3/(1.5*660e3);
+              double slope_distance_max = 20e3/(2.5*100e3);
+              double sink_velocity_min = 0.025; // m/yr
+              double sink_velocity_max = 0.10; // m/yr
+              /* minimum temperature parameters */
+              double slope_temperature_min = 250/(2.5*100*1e3); // deg/m (2.5 because dip in ref. model is 20-25 deg)
+              double slope_temperature_max = 500/(2.5*100*1e3); // deg/m
+              double temperature_min = 750;
+              double temperature_max = 950;
```

Comment:
The values above not changed and know at compile time, so you should be able to make them `constexpr` (`constexpr double offset_distance_min = 25*1e3;`.

Reply:
Done.